docs/html/development/architecture/configuration-files.rst

```diff
@@ -0,0 +1,89 @@
+===========================
+Configuration File Handling
+===========================
+
+The ``pip._internal.configuration`` module is responsible for handling
+configuration files (eg. loading from and saving values to) that are used by
+pip. The module's functionality is largely exposed through and coordinated by
+the module's ``Configuration`` class.
+
+.. note::
+
+    This section of the documentation is currently being written. pip
+    developers welcome your help to complete this documentation. If you're
+    interested in helping out, please let us know in the
+    `tracking issue <https://github.com/pypa/pip/issues/6831>`_.
```

Comment:
I see this is being repeated in #7859. Would it make sense to make this a custom derivative? This can be useful for other parts of the documentation as well, if the tracking issue URL is made variable.

Reply:
Yup - that's an excellent idea. I'll do it in a separate PR tho.